Improve printOccurrences function with streaming output for large SBO…

deeshantk opened the pull request:
Refactored the printOccurrences function to use streaming for better performance with large SBOM datasets.

prabhu commented:
This is amazing. Could you kindly run pnpm lint and remove anybextra new lines. Also revert the package.json changes.

deeshantk commented:
> This is amazing. Could you kindly run pnpm lint and remove any extra new lines. Also revert the package.json changes.

sure

deeshantk commented:
> This is amazing. Could you kindly run pnpm lint and remove any extra new lines. Also revert the package.json changes.

I have run pnpm lint, removed any extra new lines, and reverted the changes to package.json. The pull request has been updated accordingly.

Please let me know if anything else is needed!

prabhu commented:
Please remove package-lock.json and .vscode changes.

deeshantk commented:
> Please remove package-lock.json and .vscode changes.

done

prabhu commented:
I meant this branch should only have display.js changes alone.

deeshantk commented:
> I meant this branch should only have display.js changes alone.

Thank you for the clarification! I have now updated the branch so that it only includes the changes to display.js

prabhu commented on lib/helpers/display.js:
```diff
@@ -178,34 +178,51 @@ const locationComparator = (a, b) => {
 export function printOccurrences(bomJson) {
-  const data = [["Group", "Name", "Version", "Occurrences"]];
   if (!bomJson || !bomJson.components) {
+    console.error("Invalid or empty SBOM data.");
```
This error log can be removed, since empty bom data will be flagged elsewhere.

prabhu commented on lib/helpers/display.js:
```diff
@@ -178,34 +178,51 @@ const locationComparator = (a, b) => {
+      width: 50,
     },
+    columnCount: 4,
+    columns: [{ width: 20 }, { width: 40 }, { width: 50 }, { width: 25 }],
```
Is this tested? Width 25 in column 4 for filenames might be too small.

deeshantk replied:
> Is this tested? Width 25 in column 4 for filenames might be too small.

I've removed the redundant error log as suggested and adjusted the width for filenames in column 4 to better accommodate longer names.
Here is how it looks:

<img width="1414" alt="Screenshot 2024-12-05 at 3 57 12 PM" src="https://github.com/user-attachments/assets/f0ed4aea-ce38-4955-9dd1-4df59219ef09">

prabhu reviewed (approved):
Thank you so much!

aryan-rajoria commented:
Fixes issue #1482.
